src/litserve/server.py

```diff
@@ -500,24 +500,40 @@ async def win_data_streamer(self, read, write):
 
     async def data_streamer(self, read: Connection, write: Connection):
         data_available = asyncio.Event()
-        while True:
-            # Calling poll blocks the event loop, so keep the timeout low
-            if not read.poll(0.001):
-                asyncio.get_event_loop().add_reader(read.fileno(), data_available.set)
+        queue = asyncio.Queue()
+        loop = asyncio.get_event_loop()
+
+        def reader():
+            try:
+                while read.poll(0.001):  # Check if there's data available to read
```

Comment:
I would shorten this time, since with this PR we will just consume all available items on one go later on.

Actually... we shouldn't pass anything to `read.poll` at all here. If we want to sleep we need to use `asyncio.sleep` since it will avoid blocking the event loop.

Reply:
right, ideally timeout is not required here since we only poll when data is available and when it's not then we break this loop. Removed the timeout!